presto-main/src/main/java/com/facebook/presto/operator/PagesIndex.java

```diff
@@ -469,4 +470,25 @@ public String toString()
                 .add("estimatedSize", estimatedSize)
                 .toString();
     }
+
+    public Iterator<Page> getPages(int[] outputChannels)
+    {
+        return new Iterator<Page>() {
+            int currentPosition = 0;
+            PageBuilder pageBuilder = new PageBuilder(types);
+            @Override
+            public boolean hasNext()
+            {
+                pageBuilder.reset();
+                currentPosition = buildPage(currentPosition, outputChannels, pageBuilder);
+                return !pageBuilder.isEmpty();
```

Comment:
Also, i looked at https://github.com/prestodb/presto/pull/8166/files#diff-8a5509b6f2f139b307b3b788ce77e5c4 and see it's not using `PageBuilder` internally, it just returns `Blocks`. Thus it looks like no copying vs copying, but maybe I'm being misled -- needs investigation.